src/core/Akka/Routing/SmallestMailbox.cs

```diff
@@ -35,7 +35,7 @@ private Routee SelectNext(Routee[] routees)
             var winningScore = long.MaxValue;
 
             // round robin fallback
-            var winner = routees[Interlocked.Increment(ref next) % routees.Length];
+            var winner = routees[(Interlocked.Increment(ref next) & int.MaxValue) % routees.Length];
```

Comment:
Looks like you're right: https://msdn.microsoft.com/en-us/library/vstudio/dd78zt0c(v=vs.100).aspx